Documentation/Design/lambda_body_design.md

```diff
@@ -0,0 +1,72 @@
+# GLS Improvement: Lambda Support
+Engineer: faulda
+
+## Feature Overview
+This improvement adds support for lambda function bodies to GLS. Sometimes called closures or anonymous functions, lambdas are unnamed, single or multi-line functions that are declared in an expression. They can be assigned to variables and passed as parameters, which makes them versatile and useful.
+
+Some languages support multi-line lambdas, or lambdas that consist of multiple code statements. Because Python does not support multi-line lambdas, GLS is not able to support them. 
+
+## Language Examples
+### Java:  
+```Java
+(p) -> p.getGender() == Person.Sex.MALE && p.getAge() >= 18 && p.getAge() <= 25
+```
+
+### Python:
+```Python
+lambda x: math.sqrt(x) 
+```
+
+### C#:
+```C#
+(x, y) => x == y;
+```
+
+### Ruby:
+```Ruby
+lambda { |x| puts x }
+->(x) { puts x }
+```
+Both of the above constructs are valid for Ruby. GLS generates the top one for no reason at all.
+
+### TypeScript
+```TypeScript
+(amount, interestRate, duration) => amount * interestRate * duration / 12
+```
+
+
+## Design
+### GLS Syntax:
+```
+lambda : <parameter 1 name> <parameter 1 type> <parameter 2 name> <parameter 2 type> ... { <GLS command> }
```

Comment:
A more standard way of phrasing this would be

`lambda` `:` *`[parameterName, parameterType, ...]`* `command`